packages/obonode/obojobo-chunks-list/viewer-component.js

```diff
@@ -42,8 +62,14 @@ const renderEl = (props, node, index, indent) => {
 			)
 		case 'element': {
 			const ElType = node.type
+			const bulletColor = pickColorFromText(node, indent)
```

Comment:
`node` is an instance of MockElement, and as you figured out it's a nested structure. There's another data structure you can query that will be easier - `props.model.modelState.textGroup` is an instance of `TextGroup`, and it will have a `TextGroupItem` for each item in the list. That data structure isn't nested - if there are four items in the list then you'll have four `TextGroupItem`s and you can get each one by doing `props.model.modelState.textGroup.get(index)`. To get the StyleableText you can do `props.model.modelState.textGroup.get(index).text`. That you could pass into `pickColorFromText`, and `pickColorFromText` just needs to dig into the style of the first character as I commented above.

Reply:
That will definitely make things easier, and safer, I'll use textGroup and might have to change the function a bit but the above comment will still help